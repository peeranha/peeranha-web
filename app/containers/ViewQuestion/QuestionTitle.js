import React from 'react';
import PropTypes from 'prop-types';

const QuestionTitle = props => (
  <h3 className="question-title">{props.title}</h3>
);

QuestionTitle.propTypes = {
  title: PropTypes.string,
};

export default QuestionTitle;
