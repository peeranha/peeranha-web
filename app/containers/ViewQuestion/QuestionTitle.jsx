import React from 'react';
import PropTypes from 'prop-types';

import Base from 'components/Base';
import H3 from 'components/H3';
import TagList from 'components/TagsList';
import QuestionCommunity from 'components/QuestionForProfilePage/QuestionCommunity';

export const QuestionTitle = ({ title, tags, communityId, communities }) =>
  title ? (
    <Base position="middle">
      <H3>{title}</H3>

      <TagList
        className="my-2"
        chosenTags={tags}
        communityId={communityId}
        communities={communities}
      >
        <QuestionCommunity
          className="my-1"
          communities={communities}
          communityId={communityId}
        />
      </TagList>
    </Base>
  ) : null;

QuestionTitle.propTypes = {
  title: PropTypes.string,
  tags: PropTypes.array,
  communityId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  communities: PropTypes.array,
};

export default React.memo(QuestionTitle);
