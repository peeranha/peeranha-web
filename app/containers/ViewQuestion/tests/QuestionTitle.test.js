import { QuestionTitle } from '../QuestionTitle';

describe('QuestionTitle', () => {
  it('test', () => {
    const props = {
      title: 'title',
    };
    expect(QuestionTitle(props)).toMatchSnapshot();
  });
});
