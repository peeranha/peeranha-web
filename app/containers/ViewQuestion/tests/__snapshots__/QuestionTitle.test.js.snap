// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionTitle test 1`] = `
<h3
  className="question-title"
>
  title
</h3>
`;
